HandsOnProject/app/src/main/java/com/example/handsonproject/MainActivity.kt

```diff
@@ -1,11 +1,159 @@
 package com.example.handsonproject
 
-import androidx.appcompat.app.AppCompatActivity
+
 import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import com.example.handsonproject.databinding.ActivityMainBinding
+import com.example.handsonproject.databinding.ActivityMainBinding.inflate
 
+enum class Operation{
+    PLUS,MINUS,MULTIPLY,DIVIDE,MAIN_STATUS
+}
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private var operation : Operation = Operation.MAIN_STATUS
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = inflate(LayoutInflater.from(this))
+        setContentView(binding.root)
+        with(binding){
+            stash.text=""
+        }
+    }
+
+
+    fun onTapB1(view: View){
+        with(binding) { input.setText(input.getText().toString() + "1") }
+    }
+    fun onTapB2(view: View){
+        with(binding) { input.setText(input.getText().toString() + "2") }
+    }
+    fun onTapB3(view: View){
+        with(binding) { input.setText(input.getText().toString() + "3") }
+    }
+    fun onTapB4(view: View){
+        with(binding) { input.setText(input.getText().toString() + "4") }
+    }
+    fun onTapB5(view: View){
+        with(binding) { input.setText(input.getText().toString() + "5") }
+    }
+    fun onTapB6(view: View){
+        with(binding) { input.setText(input.getText().toString() + "6") }
+    }
+    fun onTapB7(view: View){
+        with(binding) { input.setText(input.getText().toString() + "7") }
+    }
+    fun onTapB8(view: View){
+        with(binding) { input.setText(input.getText().toString() + "8") }
+    }
+    fun onTapB9(view: View){
+        with(binding) { input.setText(input.getText().toString() + "9") }
+    }
+    fun onTapB0(view: View){
+        with(binding) { input.setText(input.getText().toString() + "0") }
+    }
+    fun onTapBEqual(view: View){
+        with(binding){
+            if (input.text.toString() == "" && stash.text.toString() ==  "" ) return
+            when(operation){
+                Operation.PLUS -> {
+                    val res=input.text.toString().toDouble()+stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
+                }
+                Operation.MINUS -> {
+                    val res=stash.text.toString().toDouble()-input.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
+                }
+                Operation.MULTIPLY -> {
+                    val res=input.text.toString().toDouble()*stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
```

Comment:
@shovian 

`stash.setText("")` seems to be the same for all cases except `MAIN_STATUS` , maybe it can be moved outside the `when` clause? please check.

Reply:
I think you resolved this